nni/algorithms/compression/pytorch/pruning/iterative_pruner.py

```diff
@@ -0,0 +1,584 @@
+        self.scale = scale
+        self._trainer = trainer
+        self._criterion = criterion
+
+    def validate_config(self, model, config_list):
+        schema = CompressorSchema([{
+            'sparsity': And(float, lambda n: 0 < n < 1),
+            'op_types': ['BatchNorm2d'],
+            Optional('op_names'): [str]
+        }], model, logger)
+
+        schema.validate(config_list)
+
+        if len(config_list) > 1:
+            logger.warning('Slim pruner only supports 1 configuration')
+
+    def _get_threshold(self):
+        weight_list = []
+        for (layer, _) in self.get_modules_to_compress():
+            weight_list.append(layer.module.weight.data.abs().clone())
+        all_bn_weights = torch.cat(weight_list)
+        k = int(all_bn_weights.shape[0] * self.config_list[0]['sparsity'])
+        self.masker.global_threshold = torch.topk(
+            all_bn_weights.view(-1), k, largest=False)[0].max()
+        print(f'set global threshold to {self.masker.global_threshold}')
+
+    def _callback(self):
+        for i, wrapper in enumerate(self.get_modules_wrapper()):
+            wrapper.module.weight.grad.data.add_(self.scale * torch.sign(wrapper.module.weight.data))
+
+
+class TaylorFOWeightFilterPruner(IterativePruner):
+    """
+    Parameters
+    ----------
+    model : torch.nn.Module
+        Model to be pruned
+    config_list : list
+        Supported keys:
+            - sparsity : How much percentage of convolutional filters are to be pruned.
+            - op_types : Currently only Conv2d is supported in TaylorFOWeightFilterPruner.
+    optimizer: torch.optim.Optimizer
+            Optimizer used to train model
+    training_epochs: int
+        The number of epochs to calculate the contributions.
+    dependency_aware: bool
+        If prune the model in a dependency-aware way. If it is `True`, this pruner will
+        prune the model according to the l2-norm of weights and the channel-dependency or
+        group-dependency of the model. In this way, the pruner will force the conv layers
+        that have dependencies to prune the same channels, so the speedup module can better
+        harvest the speed benefit from the pruned model. Note that, if this flag is set True
+        , the dummy_input cannot be None, because the pruner needs a dummy input to trace the
+        dependency between the conv layers.
+    dummy_input : torch.Tensor
+        The dummy input to analyze the topology constraints. Note that, the dummy_input
+        should on the same device with the model.
+
+    """
+
+    def __init__(self, model, config_list, optimizer, trainer, criterion, training_epochs=1,
+                 dependency_aware=False, dummy_input=None):
+        super().__init__(model, config_list, optimizer=optimizer,         pruning_algorithm='taylorfo', trainer=trainer, criterion=criterion, training_epochs=training_epochs, dependency_aware=dependency_aware, dummy_input=dummy_input)
```

Comment:
remove space and add `Parameters` comments, the following pruner comments also need `Parameters` comments.